### DISCUSSION TITLE
Mehrere Prognosedienste abrufen und in Statistik aufnehmen + Verrechnungslogik

### DISCUSSION THREAD

**Discussion by RienSte** [wiggal/GEN24_Ladesteuerung#104] [Ideas]
Guten Morgen @wiggal 

ich habe festgestellt, dass bei mir keiner der Prognoseanbieter wirklich zufriedenstellende Ergebnisse liefert. Immer wieder ist einer mal (viel) zu hoch oder (viel) zu niedrig.

Folgende Idee:
Abrufen von allen 3 Prognoseanbietern (falls gewünscht und konfiguriert), speichern der Daten in separate Files, inkl. Abweichung der Prognose von der Produktion + kumulierte Fehlerquadratsumme. Damit könnte man über einen längeren Zeitraum den Anbieter rausfinden, der in Summe den geringsten Fehler liefert.

Darüber hinaus bekommt der User die Möglichkeit, auszuwählen, ob man einen bestimmten Anbieter für die Ladeberechnung nehmen möchte, oder zB aus der aktuellen Prognose das Minimum, Maximum oder den Mittelwert.

Was denkst du darüber? Ist sicher keine kleine Änderung, weil man wieder einige config-Parameter und evtl. sogar einen neuen Tab braucht. Aber für die Langzeitstatistik wäre das schon nützlich und auf jeden Fall interessant.

SG Stephan

**Comment by wiggal** [comment]
> ich habe festgestellt, dass bei mir keiner der Prognoseanbieter wirklich zufriedenstellende Ergebnisse liefert. Immer wieder ist einer mal (viel) zu hoch oder (viel) zu niedrig.

Also ich bin mir meiner Prognose (solcast) sehr zufrieden, pass immer sehr gut.

> Folgende Idee: Abrufen von allen 3 Prognoseanbietern (falls gewünscht und konfiguriert), speichern der Daten in separate Files, inkl. Abweichung der Prognose von der Produktion + kumulierte Fehlerquadratsumme. Damit könnte man über einen längeren Zeitraum den Anbieter rausfinden, der in Summe den geringsten Fehler liefert.

Die Prognosen der Anbieter in separate Files zu speichern geht schon (in weather.ini festzulegen), Auswertung muss man halt dann in einem Calctool machen

> Darüber hinaus bekommt der User die Möglichkeit, auszuwählen, ob man einen bestimmten Anbieter für die Ladeberechnung nehmen möchte, oder zB aus der aktuellen Prognose das Minimum, Maximum oder den Mittelwert.

> Was denkst du darüber? Ist sicher keine kleine Änderung, weil man wieder einige config-Parameter und evtl. sogar einen neuen Tab braucht. Aber für die Langzeitstatistik wäre das schon nützlich und auf jeden Fall interessant.

Den Bedarf sehe ich aktuell nicht so sehr, und wenn, kann ich das aktuell nicht leisten.

Gruß Wiggal

**Reply by RienSte** [reply]
Alles klar, zur Kenntnis genommen 😊
Versteh schon, dass das mit dem Ursprungsgedanken des Tools nicht mehr viel gemein hat und dass es kein kleiner Aufwand ist.
